keras_nlp/src/models/vgg/vgg_backbone.py

```diff
@@ -0,0 +1,163 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import keras
+from keras import layers
+
+from keras_nlp.src.api_export import keras_nlp_export
+from keras_nlp.src.models.backbone import Backbone
+
+
+@keras_nlp_export("keras_nlp.models.VGGBackbone")
+class VGGBackbone(Backbone):
+    """
+    This class represents Keras Backbone of VGG model.
+
+    This class implements a VGG backbone as described in [Very Deep
+    Convolutional Networks for Large-Scale Image Recognition](
+    https://arxiv.org/abs/1409.1556)(ICLR 2015).
+
+    Args:
```

Comment:
Should we add `stackwise_filters` too? Just looking at resnet. https://keras.io/api/keras_cv/models/backbones/resnet_v2/

Would allow us to pull some magic numbers out of code and into the config on kaggle. Also will allow us to test smaller version of the model (which is always good).

So VGG16:
```
stackwise_num_repeats: [2, 2, 3, 3, 3]
stackwise_filters:  [64, 64, 64, 64, 512]
```

Reply:
oops i didn't see that filters doubles each time, so maybe its. `stackwise_filters: [64, 128, 256, 512, 512]`

anyway, not something to solve here, that would go in our kaggle upload.